android/java/org/chromium/chrome/browser/crypto_wallet/util/Utils.java

```diff
@@ -1532,10 +1535,27 @@ public static SpannableString createSpannableString(
             String text, ClickableSpan clickListener, int startIndex, int endIndex, int flags) {
         assert null != text;
         SpannableString spannableString = new SpannableString(text);
-        spannableString.setSpan(clickListener, startIndex, endIndex, flags);
+        if (startIndex >= 0 && endIndex > startIndex && endIndex < text.length())
+            spannableString.setSpan(clickListener, startIndex, endIndex, flags);
         return spannableString;
     }
 
+    public static Spannable createSpannableUsingHTML(
+            Context context, @StringRes int resId, View.OnClickListener onClickListener) {
+        String htmlString =
+                String.format(context.getResources().getString(resId), "<a href=\"\">", "<\\a>");
+        Spannable spannable = new SpannableString(AndroidUtils.formateHTML(htmlString));
+        URLSpan[] spans = spannable.getSpans(0, spannable.length(), URLSpan.class);
+        for (URLSpan urlSpan : spans) {
+            CustomURLSpan linkSpan = new CustomURLSpan(urlSpan.getURL(), onClickListener);
+            int spanStart = spannable.getSpanStart(urlSpan);
+            int spanEnd = spannable.getSpanEnd(urlSpan);
+            spannable.setSpan(linkSpan, spanStart, spanEnd, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
+            spannable.removeSpan(urlSpan);
+        }
```

Comment:
add docs for why we need `createSpannableUsingHTML` and when this should be used, probably use a more meaningful name as this seems to be fixing a specific case?

Reply:
adding block comment before method for document